File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/DeleteColumn.java
package com.antelopesystem.crudframework.crud.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Used to define the boolean column when using {@link Deleteable#softDelete()}
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface DeleteColumn {

}

File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/crud/annotation/DeleteColumn.kt
package com.antelopesystem.crudframework.crud.annotation

/**
 * Used to define the boolean column when using [Deleteable.softDelete]
 * ```
 * Example usage:
 *
 * ```
 * @Deleteable(softDelete = true)
 * class Entity {
 *  ...
 *  @DeleteColumn
 *  var isDeleted: Boolean = false
 * }
 * ```
 */
@Target(AnnotationTarget.FIELD)
annotation class DeleteColumn 